### ISSUE TITLE
feature: customize OUTPUT_GROUP_MAX_ENTRIES

### ISSUE THREAD

**Issue by kallewoof** [bitcoin/bitcoin#17854]
Currently the max number of UTXO:s inside one address group is hard coded as 10. In certain cases, people may want to tweak this (e.g. if they have 11+ outputs that all go to the same address and they wanna group all of them together so they are guaranteed to be spent as one, they can't do that right now without recompiling the code).

As a side note, this value is only there to prevent too-large transactions due to tons of address reuse in one wallet -- I think 10 is way too low for this case. Thoughts, @sdaftuar?

**Comment by fjahr** [comment]
> Currently the max number of UTXO:s inside one address group is hard coded as 10. In certain cases, people may want to tweak this (e.g. if they have 11+ outputs that all go to the same address and they wanna group all of them together so they are guaranteed to be spent as one, they can't do that right now without recompiling the code).

I agree that there should be something that makes this easier than constructing a raw transaction by hand but I am not sure this is the best solution for this particular problem. Even if `OUTPUT_GROUP_MAX_ENTRIES` is increased there is no guarantee that the user will actually get the address (-group) they want out of coin selection if there are other outputs in the wallet. I would rather suggest adding a `spend_from` option to `fundrawtransaction` and/or `sendtoaddress`. It would take a list of addresses that should be used to fund the tx until the tx it is fully funded. There could also be also `spend_all_from` which would use all the inputs from all the addresses in the list. I will look into this implementing this.

**Comment by decryp2kanon** [comment]
> I think 10 is way too low for this case.
agreed

**Comment by pinheadmz** [comment]
This is no longer an issue in recent versions of Bitcoin Core. We'll close it for now, please feel free to add additional comments or open a new issue if you think that would be helpful. 

The new limit is 100:
https://github.com/bitcoin/bitcoin/blob/904a98702e7372ed186e3faa81ef64f1ce49c945/src/wallet/spend.cpp#L33
